string.c

```diff
@@ -4134,9 +4134,12 @@ rb_str_hash_cmp(VALUE str1, VALUE str2)
  *   hash -> integer
  *
  * Returns the integer hash value for +self+.
- * The value is based on the length, content and encoding of +self+.
  *
- * Related: Object#hash.
+ * Two hashes with the same content and encoding have the same hash value
+ * (and will compare using #eql?);
```

Comment:
What will compare?

Reply:
Actually, I thought the same thing.  I borrowed this language from elsewhere (Array, Complex, and Struct), but will revise.  I've also made a note to myself to rationalize the language in all #hash methods.

Reply:
Revised and expanded.